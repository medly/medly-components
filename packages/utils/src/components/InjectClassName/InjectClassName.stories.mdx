import { Meta } from '@storybook/addon-docs/blocks';

<Meta title="Utils/Components/InjectClassName" />

# InjectClassName

`InjectClassName` can be used to create styles for child elements without providing the html tag to styled-components.

#### Example

```tsx
import { defaultTheme } from '@medly-components/theme';
import { InjectClassName } from '@medly-components/utils';
import styled from 'styled-components';
import { SvgLoaderProps } from './types';

export const SvgLoaderStyled = styled(InjectClassName)<SvgLoaderProps>`
    font-size: ${({ theme, size }) => theme.loader.sizes[size!]};
    * {
        fill: ${({ theme, color }) => color || theme.loader.defaultColor};
    }
`;

SvgLoaderStyled.displayName = 'SvgLoader';
SvgLoaderStyled.defaultProps = {
    size: 'S',
    theme: defaultTheme
};
```

Now the above SvgLoader can be used to style any loader as below:

```jsx
<SvgLoader>
    <ConcentricCircleLoaderSvg />
</SvgLoader>
```

**Note: Here, the SvgLoader style applies directly to ConcentricCircleLoaderSvg.**
