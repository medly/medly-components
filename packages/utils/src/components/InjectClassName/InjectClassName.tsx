import React from 'react';
import { Props } from './types';

const InjectClassName: React.FunctionComponent<Props> = props => {
    const { className, children } = props;
    return (
        <>
            {React.Children.map(children, child => {
                return React.cloneElement(child as any, {
                    className
                });
            })}
        </>
    );
};

export default InjectClassName;
