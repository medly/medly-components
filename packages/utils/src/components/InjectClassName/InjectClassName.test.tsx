import { render } from '@test-utils';
import InjectClassName from './InjectClassName';

describe('InjectClassName component', () => {
    it('should inject className prop to all children', () => {
        const { container } = render(
            <InjectClassName className="test-className">
                <div>Test div 1</div>
                <div>Test div 2</div>
            </InjectClassName>
        );
        expect(container).toMatchSnapshot();
    });
});
