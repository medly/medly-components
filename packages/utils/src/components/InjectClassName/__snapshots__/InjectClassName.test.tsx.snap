// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InjectClassName component should inject className prop to all children 1`] = `
<div>
  <div
    class="test-className"
  >
    Test div 1
  </div>
  <div
    class="test-className"
  >
    Test div 2
  </div>
</div>
`;
